File: problem-solving/baekjoon/15828_Router.go
// https://www.acmicpc.net/problem/15828
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int
	fmt.Fscanln(reader, &n)
	var queue []int
	for {
		var input int
		fmt.Fscanln(reader, &input)
		if input == -1 {
			break
		} else if input == 0 {
			queue = queue[1:]
		} else {
			if len(queue) < n {
				queue = append(queue, input)
			}
		}
	}
	if len(queue) == 0 {
		fmt.Fprintln(writer, "empty")
	} else {
		for i := 0; i < len(queue); i++ {
			fmt.Fprintf(writer, "%d ", queue[i])
		}
	}
}

File: problem-solving/baekjoon/15828_Router.py
# https://www.acmicpc.net/problem/15828
import sys
import queue

if __name__ == "__main__":
    n = int(sys.stdin.readline())
    q = queue.Queue()
    while True:
        input = int(sys.stdin.readline())
        if input == -1:
            break
        elif input == 0:
            q.get()
        else:
            if q.qsize() < n:
                q.put(input)
    if q.qsize() == 0:
        print("empty")
    else:
        while not q.empty():
            print("{} ".format(q.get()), end='')